### ISSUE TITLE
clock_gettime() not available on osx

### ISSUE THREAD

**Issue by wiedi** [OLSR/olsrd#4]
In b869571f339972550dbde7c0341e3db85fdf5ea1 ("scheduler: use the monotonic clock") calls to `clock_gettime` were introduced which are not available on mac os x:

```
src/scheduler.c:134:7: warning: implicit declaration of function 'clock_gettime' is invalid in C99 [-Wimplicit-function-declaration]
  if (clock_gettime(CLOCK_MONOTONIC, &tv) != 0) {
      ^
src/scheduler.c:134:21: error: use of undeclared identifier 'CLOCK_MONOTONIC'
  if (clock_gettime(CLOCK_MONOTONIC, &tv) != 0) {
                    ^
src/scheduler.c:631:21: error: use of undeclared identifier 'CLOCK_MONOTONIC'
  if (clock_gettime(CLOCK_MONOTONIC, &first_tv)) {
```

Would bringing the old code back when `defined __APPLE__` be a viable option? I might be able to put a PR together if this sounds reasonable.

**Comment by fhuberts** [comment]
sure, go for the PR

**Comment by bittorf** [comment]
maybe this can help:
https://github.com/br101/horst/pull/54/commits/7e51f1670b8e21eddcca0b3d4e76bfa089f3c23e

**Comment by wiedi** [comment]
this is basically the same approach used by my latest revision

**Comment by bittorf** [comment]
wiedi: fine, please try to cook a patch - so it can get into the release...

**Comment by wiedi** [comment]
@bittorf oh, sorry I wasn't clear. It's already waiting for review at https://github.com/OLSR/olsrd/pull/5

**Comment by fhuberts** [comment]
fixed by e554f4f96d14263419b7576790fe55536397d50c
